pinot-core/src/main/java/org/apache/pinot/core/operator/combine/function/CombineFunction.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.pinot.core.operator.combine.function;
+
+import org.apache.pinot.core.operator.blocks.results.BaseResultsBlock;
+
+
+public interface CombineFunction<T extends BaseResultsBlock> {
+
+  /**
+   * Merges a results block into the main mergeable results block.
+   * <p>NOTE: {@code blockToMerge} should contain the result for a segment without any exception. The errored segment
+   * result is already handled.
+   *
+   * @param mergedBlock The block that accumulates previous results. It should be modified to add the information of the
+   *                    other block.
+   * @param blockToMerge The new block that needs to be merged into the mergedBlock.
+   */
+  void mergeResultsBlocks(T mergedBlock, T blockToMerge);
+
+  /**
+   * Can be overridden for early termination. The input results block might not be mergeable.
+   */
+  default boolean isQuerySatisfied(T resultsBlock) {
```

Comment:
nit: consider naming this `shouldEarlyTerminate`?

Reply:
shouldEarlyTerminate also can mean exception early terminate. i think here it is specific to query is satisfied so we early terminate. anyway I added javadoc